examples/sssp/sssp.cu

```diff
@@ -1,44 +1,73 @@
 #include <cstdlib>  // EXIT_SUCCESS
 
-#include <gunrock/applications/sssp/sssp.hxx>
+#include <gunrock/applications/runner.hxx>
+#include <gunrock/applications/sssp/sssp_implementation.hxx>
 
 using namespace gunrock;
 using namespace memory;
 
 void test_sssp(int num_arguments, char** argument_array) {
-  using vertex_t = int;
-  using edge_t = int;
-  using weight_t = float;
-
+  
   if (num_arguments != 2) {
-    std::cerr << "usage: ./bin/color filename.mtx" << std::endl;
+    std::cerr << "usage: ./bin/<program-name> filename.mtx" << std::endl;
     exit(1);
   }
-
-  // Load Matrix-Market file & convert the resultant COO into CSR format.
+  
+  // --
+  // Define types
+  
+  using vertex_t = int;
+  using edge_t   = int;
+  using weight_t = float;
+  
+  // --
+  // IO
+  
   std::string filename = argument_array[1];
+  
   io::matrix_market_t<vertex_t, edge_t, weight_t> mm;
-  auto coo = mm.load(filename);
-
-  // convert coo to csr
-  format::csr_t<memory_space_t::device, vertex_t, edge_t, weight_t> csr;
-  csr = coo;  // Able to convert host-based coo_t to device-based csr (or host
-              // to host). As of right now, it requires coo to be host side.
+  format::csr_t<memory::memory_space_t::device, vertex_t, edge_t, weight_t> csr;
+  csr.from_coo(mm.load(filename)); 
 
-  vertex_t source = 0;
-  thrust::device_vector<weight_t> d_distances(csr.number_of_rows);
+  // --
+  // Build graph + metadata
+  
+  auto [G, meta] = graph::build::from_csr_t<memory_space_t::device>(&csr);
```

Comment:
I think this bit, we should change it to just be:
```cpp
auto [G, meta] = graph::build<memory_space_t::device>(&coo);
```
and somehow specify which views you want. So, maybe with flags?
```cpp
auto [G, meta] = graph::build<memory_space_t::device, graph_view_t::csr, graph_view_t::coo ...>(&coo);
```
And it will build `G` with all the views, without having to do the format conversion of `coo` to `csr` yourself.

Reply:
Maybe let's leave this til we have a use-case for multiple / non-csr graph types?  I haven't seen a case where a graph gets built w/ something else, so not sure exactly how you're intending on setting up the datastructures.